MovingBlocks/Terasology
# Change advanced game create UI to clarify the game creation process

### Contains

Changes button positions, widget locations, and adds/changes labels to make the game creation process easier to understand for the player. In addition, this PR also resolves two points in #3400.

### List of UI Changes

- **Larger changes**:
  - Navigation buttons from universe setup onwards are now in the margins of the screen, instead of the bottom.
  - All navigation buttons from new game onwards are now in the consistant form _return-[play]-continue_, going from left to right.
  - All navigation buttons have had their labels updated to describe the screen they direct to.
  - All Navigation buttons have been made significantly larger.
- **Smaller changes**:
  - All advanced game creation UI containers have been anchored to the center of the screen, so side buttons are at the same height in every screen.
  - All screens with world previews now have the UI anchored to keep the preview area square.
  - "Module Details" has been moved to be underneith the module description.
  - Module filters are now anchored to be over the 'reset filters' button.
  - All inactive "play" buttons have been removed.
  - Added a missing 'subtitle' label in spawn preview, resolving an error in the console.
- **Changes to resolve some points in #3400**:
  - Disclaimers have been added for incomplete multi-world support and the loading time of the module screen.
  - "Seed" label moved into the same line as the seed field.

### Screenshots

New Game
![newgamefinal](https://user-images.githubusercontent.com/44126397/49334546-3f9cec80-f5b3-11e8-8e2d-8be136e80674.PNG)

Advanced Game/Module Select
![advancedgamefinal](https://user-images.githubusercontent.com/44126397/49334548-4af01800-f5b3-11e8-9e87-93b1f18a8592.PNG)

Universe Setup
![universesetupfinal](https://user-images.githubusercontent.com/44126397/49334549-4d527200-f5b3-11e8-9f37-3aea16048562.PNG)

Pregeneration
![pregenerationfinal](https://user-images.githubusercontent.com/44126397/49334551-53485300-f5b3-11e8-8bbe-c0ad52f2997a.PNG)

Spawn Preview
![spawnpreviewfinal](https://user-images.githubusercontent.com/44126397/49334552-55121680-f5b3-11e8-9a6f-1a26b08be936.PNG)

### How to test

- Create a game using the 'advanced' button in the new game menu, all the UI changes should work as expected.
- Try resizing the screen, and see if some widgets break their bounding boxes.

### Outstanding

- [x] Add translation strings (I'm waiting to see if any changes need to be made before I add these)

### Other things

- I avoided putting the buttons on the margins of the module select screen, since that could be too large for smaller screens with three columns. I made a version of the module selection screen with only two columns instead of three [here](https://github.com/AndyTechGuy/Terasology/tree/twoColumnModuleSelect), would it be better to replace the current screen with this so the buttons can fit?
  - Two columns looks like this:
![advancedgamesetup](https://user-images.githubusercontent.com/44126397/49334574-b1753600-f5b3-11e8-94f7-b2d1a1c1bd49.PNG)

- The navigation buttons in the new game screen are not as large as the advanced creation screens, should this be changed?
- Any feedback/suggestions on how to improve the UI look further is greatly appreciated!
